src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -367,36 +367,27 @@ internal void Free(SyntaxListPool pool)
 
         internal CompilationUnitSyntax ParseCompilationUnit()
         {
+            return ParseWithStackGuard(
+                ParseCompilationUnitCore,
+                () => SyntaxFactory.CompilationUnit(
+                        new SyntaxList<ExternAliasDirectiveSyntax>(),
```

Comment:
The `ParseWithStackGuard` method is responsible for attaching the error.  This call back is providing the node on which it will attach the error.